api/envoy/api/v2/route/route.proto

```diff
@@ -903,6 +903,26 @@ message RetryPolicy {
     google.protobuf.Duration max_interval = 2 [(validate.rules).duration = {gt {}}];
   }
 
+  message RetryBudget {
+    // Specifies the limit on concurrent retries as a percentage of the sum of active requests,
+    // active connections, and active pending requests.
+    //
+    // .. note::
+    //
+    //    The approach of summing active requests and connections works because the active request
+    //    count for HTTP/1.1 and HTTP/2 uses either active connection or active request counts
+    //    respectively. However, Envoy does not reclaim HTTP/1.1 connections until it is required,
+    //    so the use of active connections is imperfect.
+    //
+    // This parameter is optional. Defaults to 20%.
+    type.Percent percent_budget = 1;
+
+    // Specifies the minimum retry concurrency allowed for the retry budget.
+    //
+    // This parameter is optional. Defaults to 10.
```

Comment:
If we do set a default here and it's arbitrary, I would just use 3 which is the current default for concurrent retry circuit breaker. I also would suggest allowing this to be pulled from runtime like we do with the static retry circuit breakers.

Reply:
What I was going for here was for the default minimum concurrency to make the allowed concurrent retries start at 3, the default for the retry circuit breaker. I actually need to bump the default to 15 since the budget is 20%.

Reply:
OK that make sense. Can you please clarify that reasoning in the docs.